common/spec/dependabot/metadata_finders/base/changelog_finder_spec.rb

```diff
@@ -165,44 +173,72 @@
-          end
+          context "and it points to a specific directory" do
+            let(:suggested_changelog_url) do
+              "github.com/mperham/sidekiq/blob/master/dir/Pro-Changes.md"
+            end
 
-          it "gets the right URL" do
-            expect(subject)
-              .to eq(
-                "https://github.com/mperham/sidekiq/blob/master/Pro-Changes.md"
-              )
+            before do
+              sidekiq_changelog =
+                fixture("github", "contents_sidekiq_changelog.json")
+              suggested_github_url =
+                "https://api.github.com/repos/mperham/sidekiq/contents/dir?ref=master"
+              stub_request(:get, suggested_github_url)
+                .to_return(status: 200,
+                           body: sidekiq_changelog,
+                           headers: { "Content-Type" => "application/json" })
+            end
+
+            it "gets the right URL" do
+              expect(subject)
+                .to eq(
+                  "https://github.com/mperham/sidekiq/blob/master/Pro-Changes.md"
+                )
+            end
           end
 
-          context "that can't be found" do
+          context "but it does not point to a specific directory" do
+            let(:suggested_changelog_url) do
+              "github.com/mperham/sidekiq/blob/master/Pro-Changes.md"
+            end
+
             before do
+              suggested_github_response =
+                fixture("github", "contents_sidekiq.json")
               suggested_github_url =
                 "https://api.github.com/repos/mperham/sidekiq/contents/"
               stub_request(:get, suggested_github_url)
-                .to_return(status: 404)
+                .to_return(status: 200,
+                           body: suggested_github_response,
+                           headers: { "Content-Type" => "application/json" })
             end
 
-            it "falls back to looking for the changelog as usual" do
+            it "gets the right URL" do
               expect(subject)
                 .to eq(
-                  "https://github.com/gocardless/business/" \
-                  "blob/master/CHANGELOG.md"
+                  "https://github.com/mperham/sidekiq/blob/master/Pro-Changes.md"
                 )
             end
           end
+
+          context "that can't be found" do
+            before do
+              suggested_github_url =
+                "https://api.github.com/repos/mperham/sidekiq/contents/"
+              stub_request(:get, suggested_github_url)
+                .to_return(status: 404)
+
+              suggested_github_url_with_tag =
+                "https://api.github.com/repos/mperham/sidekiq/contents/?ref=v1.4.0"
+              stub_request(:get, suggested_github_url_with_tag)
+                .to_return(status: 404)
+            end
+
+            it "returns nil for the changelog url" do
+              expect(subject).to eq(nil)
+            end
```

Comment:
I thought about showing that the ChangelogFinder will still try to find the changelog through other means, but there are other tests for that ("but with a changelog on the tag"), and I decided that returning nil instead of throwing an error was the important thing to test